dynamic_programming/special_numbers.cpp

```diff
@@ -0,0 +1,218 @@
+/**
+ * @file
+ * @brief A positive number is special if all of its digits are distinct.
+ * The purpose of this problem is to find out all the special numbers
+ * in range [1,n]
+ *
+ * @details
+ * This problem can be classified under Digit DP. The idea is to use bitmask and
+ * Dynamic Programmic to optimize our solution for the larger values of n.
+ * For example: example 1:- N: 20, OUTPUT: 19
+ *
+ * For example: example 2:-
+ * X: '5', OUTPUT: 2
+ * @author [Maaz](https://github.com/riddle07)
+ * @see more on
+ * https://leetcode.com/problems/count-special-integers/description/
+ */
+
+// header files
+#include <algorithm>
+#include <array>
+#include <bitset>
+#include <cassert>
+#include <chrono>
+#include <climits>
+#include <cmath>
+#include <complex>
+#include <cstring>
+#include <functional>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <numeric>
+#include <queue>
+#include <random>
+#include <set>
+#include <string>
+#include <vector>
+/**
+ * @namespace dynamic_programming
+ * @brief Digit Dynamic Programming Algorithms
+ */
+namespace dynamic_programming {
+
+/**
+ * @namespace special_numbers
+ * @brief Counting Special Numbers in range [1,n]
+ */
+namespace special_numbers {
+
+// vector for storing the upper limit for each digit place
+
+std::vector<int> v;
+
+// dp array to memoize the solution for given transition states
+
+int dp[(1 << 10)][11][2][2];
+/**
+ * Depth-first-Search Function to check for the eligibility for digits
```

Comment:
Could you split this documentation into the following tags
@brief
@details
@param <parameter name> <parameter description>